### PR TITLE
pacific: rgw: rgw_parse_url_bucket() rejects empty bucket names after 'tenant:'

### PR DESCRIPTION
**Pull request by cbodley**
backport tracker: https://tracker.ceph.com/issues/58275

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_bucket.h [hunk 1]
```diff
@@ -49,9 +49,10 @@ extern int rgw_bucket_parse_bucket_key(CephContext *cct, const string& key,
 extern std::string rgw_make_bucket_entry_name(const std::string& tenant_name,
                                               const std::string& bucket_name);
 
-extern void rgw_parse_url_bucket(const string& bucket,
-                                 const string& auth_tenant,
-                                 string &tenant_name, string &bucket_name);
+[[nodiscard]] int rgw_parse_url_bucket(const std::string& bucket,
```
Comment (ivancich) [inline src/rgw/rgw_bucket.h:52]: I hadn't known about `[[nodiscard]]` previously. Cool!

**Review by ivancich — APPROVED** [review]
LGTM

**Comment by cbodley** [comment]
jenkins test api

**Comment by cbodley** [comment]
jenkins test make check

**Comment by ljflores** [comment]
Rados suite review:

https://pulpito.ceph.com/?branch=wip-yuri3-testing-2023-04-04-0833-pacific

Failures, unrelated:
    1. https://tracker.ceph.com/issues/49287
    2. https://tracker.ceph.com/issues/53827
    3. https://tracker.ceph.com/issues/55347
    4. https://tracker.ceph.com/issues/57386
    5. https://tracker.ceph.com/issues/54071
    6. https://tracker.ceph.com/issues/58585
    7. https://tracker.ceph.com/issues/59529
    8. https://tracker.ceph.com/issues/59530

Details:
    1. podman: setting cgroup config for procHooks process caused: Unit libpod-$hash.scope not found - Ceph - Orchestrator
    2. cephadm exited with error code when creating osd: Input/Output error. Faulty NVME? - Ceph - Infrastructure - Sepia
    3. SELinux Denials during cephadm/workunits/test_cephadm - Ceph - Orchestrator
    4. cephadm/test_dashboard_e2e.sh: Expected to find content: '/^foo$/' within the selector: 'cd-modal .badge' but never did - Ceph - Mgr - Dashboard
    5. rados/cephadm/osds: Invalid command: missing required parameter hostname(<string>) - Ceph - Orchestrator
    6. rook: failed to pull kubelet image - Ceph - Orchestrator
    7. mds_upgrade_sequence: overall HEALTH_ERR 1 filesystem with deprecated feature inline_data; 1 filesystem is offline; 1 filesystem is online with fewer MDS than max_mds - Ceph - CephFS
    8. mgr-nfs-upgrade: mds.foofs has 0/2 - Ceph - CephFS

**Pull request merged**